pkg/cache/cache.go

```diff
@@ -172,6 +172,17 @@ type Options struct {
 	// is "done" with an object, and would otherwise not requeue it, i.e., we
 	// recommend the `Reconcile` function return `reconcile.Result{RequeueAfter: t}`,
 	// instead of `reconcile.Result{}`.
+	//
+	// SyncPeriod will trigger update events with the old object being equal to the new
+	// object, except when the cache was out of sync.
```

Comment:
> except when the cache was out of sync

I'm not sure that this is true. Do we have evidence that informers every trigger update events with old object != new object during resync?

Reply:
(Simplified version) Every syncPeriod the informer takes all objects that are stored locally and sends out update events for them

Reply:
I thought that this is a sync between the api-server and the local cache :-)

Reply:
Nope :) But that's a very frequent misunderstanding

Reply:
Re-reading the rest of the godoc above. While it contains a lot of correct information I can see how it's misleading